ui/src/app/views/concept-homepage/component.spec.ts

```diff
@@ -71,12 +72,89 @@ describe('ConceptHomepageComponent', () => {
         { provide: ActivatedRoute, useValue: activatedRouteStub }
       ]}).compileComponents().then(() => {
         fixture = TestBed.createComponent(ConceptHomepageComponent);
+        // This tick initializes the component.
         tick();
+        // This finishes the API calls.
+        updateAndTick(fixture);
+        // This finishes the page reloading.
+        updateAndTick(fixture);
       });
   }));
 
 
   it('should render.', fakeAsync(() => {
     expect(fixture).toBeTruthy();
   }));
+
+  it('should have one card per domain.', fakeAsync(() => {
+    expect(fixture.debugElement.queryAll(By.css('.card.item-card')).length)
+      .toBe(DomainStubVariables.STUB_DOMAINS.length);
+  }));
+
+  it('should default to standard concepts only, and performs a full search', fakeAsync(() => {
+    const spy = spyOn(TestBed.get(ConceptsService), 'searchConcepts')
+      .and.callThrough();
+
+    const searchTerm = 'test';
+
+    simulateInput(fixture, fixture.debugElement.query(By.css('#concept-search-input')), searchTerm);
+    simulateClick(fixture, fixture.debugElement.query(By.css('.btn-search')));
+    updateAndTick(fixture);
+
+    DomainStubVariables.STUB_DOMAINS.forEach((domain) => {
+      const expectedRequest = {
+        query: searchTerm,
+        // Tests that it searches only standard concepts.
+        standardConceptFilter: StandardConceptFilter.STANDARDCONCEPTS,
+        domain: domain.domain
+      };
+      expect(spy).toHaveBeenCalledWith(
```

Comment:
opt: I'm on the fence for these verifications. It does seem that a slightly better test would look at the rendered output and assert that it matches expectations w.r.t. STANDARDCONCEPTS. However, this might make the test case a lot less readable because the stubbed concepts are defined in another file and you shouldn't have to go read that file to understand the test. This is why I'm not super bullish on shared test models (exception: utilities / base models / cases where you just need any valid model).

A middle ground might be to have exported named CONCEPTs on the StubVariablesClass, then reference them here. e.g. check standardconcepts, then assert that row1 contains STANDARD_CONDITION_1.name.
 
Anyways - I think what you have right now is not bad so I don't object to keeping it as is. Mostly food for thought.

Reply:
That makes sense. I actually think I disagree though, because we don't really want to test the API behavior here, just that when the UI makes calls, it does so as the API expects it to be made, and then I think we want to validate that the API returns the correct values back. Otherwise I think we should add more checks, rather than removing this.